apps/relayer/src/transaction/transaction.service.ts

```diff
@@ -75,18 +87,39 @@ export class TransactionService implements OnModuleInit, OnModuleDestroy {
     }
   }
 
-  async submitTransactionObject(txo: TransactionObject<any>): Promise<Result<string, any>> {
+  async submitTransactionObject(
+    txo: TransactionObject<any>
+  ): Promise<Result<string, any>> {
     return this.submitTransaction(toRawTransaction(txo))
   }
 
   private async checkTransactions() {
     const txs = await Promise.all(
-      [...this.watchedTransactions].map(txHash => this.kit.web3.eth.getTransaction(txHash))
+      [...this.watchedTransactions].map(txHash =>
+        this.kit.web3.eth.getTransaction(txHash)
+      )
     )
 
     const completed = txs.filter(tx => tx.blockHash !== null)
-    completed.forEach(tx => this.unwatchTransaction(tx.hash))
-    const expired = txs.filter(tx => tx.blockHash == null && this.isExpired(tx.hash))
+
+    for (const tx of completed) {
```

Comment:
Let's refactor this a bit cose now it's going to wait do sequencial request to node because of the for loop.
I'd do something like:
```
await Promise.all(completed.map(tx => this.finalizeTransaction(tx))
```

And then the finalizeTransaction should first unwatch the transaction and then wait for the receipt and log.